File: prototype/ui/state.py
"""Zustandsdefinitionen fuer die Pill-Benutzeroberflaeche."""

from enum import Enum


class PillState(Enum):
    """Repraesentiert die diskreten Zustaende des schwebenden Pill-Fensters."""

    IDLE = "idle"              # Startbereit / Ruhezustand
    RECORDING = "recording"    # State 1: Aufnehmen (REC, Waveform, Timer)
    PROCESSING = "processing"  # Inferenz laeuft (Transkription im Hintergrund)
    READY = "ready"            # State 2: Kopiert (Bestaetigung, Details-Button)
    EXPANDED = "expanded"      # State 3: Ausgeklappt (Volltextpruefung & Editieren)

File: app/src/state.rs
//! Zustandsmodell fuer die schwebende whisper-pill Kapsel.

use serde::{Deserialize, Serialize};

/// Repraesentiert die fuenf Zustaende der schwebenden Desktop-Pill.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum PillState {
    /// Ruhezustand am oberen Bildschirmrand (420x48 px, 0 % CPU-Last).
    Idle,

    /// Aktive Audio-Aufnahme mit animierter Waveform (420x48 px).
    Recording,

    /// Whisper-Inferenz laeuft im Hintergrund (420x48 px).
    Processing,

    /// Transkription abgeschlossen & Text im Clipboard bereit (420x48 px).
    Ready,

    /// Ausgeklappter Editor zur Ansicht und manuellen Bearbeitung (420x240 px).
    Expanded,
}

impl Default for PillState {
    fn default() -> Self {
        PillState::Idle
    }
}
